src/components/views/dialogs/SlidingSyncOptionsDialog.tsx

```diff
@@ -27,89 +27,90 @@ import { useAsyncMemo } from "../../../hooks/useAsyncMemo";
     const id = window.setTimeout(() => controller.abort(), 10 * 1000); // 10s
     const res = await fetch(endpoint + "/client/server.json", {
         signal: controller.signal,
     });
     clearTimeout(id);
     if (res.status != 200) {
-        throw new Error(`proxyHealthCheck: proxy server returned HTTP ${res.status}`);
-    }
-    const body = await res.json();
-    if (body.server !== hsUrl) {
-        throw new Error(`proxyHealthCheck: client using ${hsUrl} but server is as ${body.server}`);
+        throw new Error(`slidingSyncHealthCheck: endpoint returned ${res.status}`);
     }
-    logger.info("sliding sync proxy is OK");
+    logger.info("slidingSyncHealthCheck: sliding sync endpoint exists");
 }
 
 export const SlidingSyncOptionsDialog: React.FC<{ onFinished(enabled: boolean): void }> = ({ onFinished }) => {
     const cli = MatrixClientPeg.safeGet();
-    const currentProxy = SettingsStore.getValue("feature_sliding_sync_proxy_url");
     const hasNativeSupport = useAsyncMemo(
         () =>
-            syncHealthCheck(cli).then(
+            nativeSlidingSyncSupportCheck(cli).then(
+                () => true,
+                () => false,
+            ),
+        [],
+        null,
+    );
+    const hasProxySupport = useAsyncMemo(
+        () =>
+            proxySlidingSyncSupportCheck(cli).then(
                 () => true,
                 () => false,
             ),
         [],
         null,
     );
 
-    let nativeSupport: string;
-    if (hasNativeSupport === null) {
-        nativeSupport = _t("labs|sliding_sync_checking");
+    let hasSupport: string;
+    if (hasNativeSupport === null && hasProxySupport === null) {
+        hasSupport = _t("labs|sliding_sync_checking");
     } else {
-        nativeSupport = hasNativeSupport
+        hasSupport = hasNativeSupport || hasProxySupport
             ? _t("labs|sliding_sync_server_support")
             : _t("labs|sliding_sync_server_no_support");
     }
 
-    const validProxy = withValidation<undefined, { error?: unknown }>({
-        async deriveData({ value }): Promise<{ error?: unknown }> {
-            if (!value) return {};
-            try {
-                await proxyHealthCheck(value, MatrixClientPeg.safeGet().baseUrl);
-                return {};
-            } catch (error) {
-                return { error };
-            }
-        },
-        rules: [
-            {
-                key: "required",
-                test: async ({ value }) => !!value || !!hasNativeSupport,
-                invalid: () => _t("labs|sliding_sync_server_specify_proxy"),
-            },
-            {
-                key: "working",
-                final: true,
-                test: async (_, { error }) => !error,
-                valid: () => _t("spotlight|public_rooms|network_dropdown_available_valid"),
-                invalid: ({ error }) => (error instanceof Error ? error.message : null),
-            },
-        ],
-    });
-
     return (
         <TextInputDialog
```

Comment:
Also, I presume I should swap this out for a different `Dialog` type. Any suggestions? :smile:

Reply:
Perhaps no dialog at all. https://github.com/matrix-org/matrix-react-sdk/pull/12307#issuecomment-1986175748